### DISCUSSION TITLE
Weapon Socket

### DISCUSSION THREAD

**Discussion by mlnant11** [Sixze/ALS-Refactored#476] [Q&A] [answered]
I'm trying to add a socket in the rifle "M4", a socket called "Muzzle", to implement an effect when its shoot.
But no matter what I do, the socket isnt found. I already verified the socket name, the skeletal mesh, but the socket is not found.

Is there something about sockets in als for unreal engine 5.3 that im missing?

Take a look on my code below:

void AAlsCharacterExample::Input_OnShoot()
{
	AAlsCharacter* CharacterInst = Cast<AAlsCharacter>(this);
	if (CharacterInst)
	{
		if (CharacterInst->GetOverlayMode() == AlsOverlayModeTags::M4)
		{
			if (CharacterInst->IsDesiredAiming())
			{
				if (FireSound)
				{
					UGameplayStatics::PlaySound2D(this, FireSound);
				}
				const USkeletalMeshSocket* BarrelSocket = GetMesh()->GetSocketByName("muzzleSocket");
				if (BarrelSocket)
				{
					UE_LOG(LogTemp, Warning, TEXT("BarrelSocket is true"));
				}
				else {
					UE_LOG(LogTemp, Warning, TEXT("BarrelSocket not found"));
				}
			}
		}
	}
}

**Comment by ameaninglessname** [comment, marked as answer]
`GetMesh()->GetSocketByName("muzzleSocket` It looks like you using the wrong mesh, ain't weapon mesh should be used here?

**Reply by mlnant11** [reply]
Im getting the weapon mesh by "GetMesh()". is it wrong?

**Reply by mlnant11** [reply]
wait, your answer made me think better... GetMesh() returns the character's mesh, not the weapon's mesh.
so I looked for my character's skeletal mesh and found a socket for that weapon there, so I just used that name socket and it worked!
Your answer helped me a lot, you made me think further.
I'm learning a little about ALS day by day, thanks my friend!
